torch_geometric/utils/get_mp_out.py

```diff
@@ -0,0 +1,49 @@
+from typing import Callable
+
+import torch
+from torch import Tensor
+
+from .subgraph import get_num_hops
+
+
+def intermediate_mp_output(i: int, model: torch.nn.Module, **kwargs) -> Tensor:
```

Comment:
I think it would be cleaner if you just return the outputs of all `MessagePassing` layers. Then you can simply do
```python
hs = get_intermediate_message_passing_embeddings(model, *args, **kwargs)
h = hs[-1]
```
Also makes implementation somewhat easier. WDYT?

Reply:
Good idea. Done.